src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -486,3 +513,14 @@ def add_snowpark_package(
                 SNOWPARK_PACKAGE_NAME,
                 ex,
             )
+
+
+def get_signature(packages: List[str]) -> str:
+    """
+    Create unique signature for a list of package names.
+    Args:
+        packages (List[str]) - A list of string package names.
+    Returns:
+        str - The signature.
+    """
```

Comment:
SHA-1 signature of alphabetically sorted set of packages. The hexadecimal value is used because this signature will be part of the zipped package file name (the signature needs to be friendly to file name conventions).